2015/proposals/schamnad-sanjeed.md

```diff
@@ -0,0 +1,115 @@
+) need to be implemented are listed as follows:
+* int TLS_accept(SSL *ssl);
+* int TLS_connect(SSL *ssl);
+* int TLS_read(SSL *ssl, void *buf, int num);
+* int TLS_write(SSL *ssl, const void *buf, int num);
+
+The output of this framework will be an encrypted stream.
+
+Clients and Peers will be authenticated using the Certificate authentication methodology, in particular by using X.509 certificates. This will be based on the methodology incorporated by OpenSSL, in particular the following functions:
+* openssl req -new -key privkey.pem -out cert.csr
+* int SSL_CTX_use_certificate(SSL_CTX *ctx, X509 *x);
+* X509 *TLS_CTX_get0_certificate(const SSL_CTX *ctx);
+* X509 *TLS_get_certificate(const SSL *ssl);
+
+3) The Application Programming Interface (API) layer
+This layer will provide high-level, abstract interfaces for the creation of application-layer protocols. As a proof of concept, HTTPS will be implemented using the API interface defined in JuliaTLS.
+
+## Schedule of Deliverables/ Milestones
+
+### May 25th -  June 7th
+Crypto.jl: will contain wrappers for functions of the libcrypto library for the �MD2, MD5, SHA, SHA1, SHA224, and SHA256 encryption algorithms.
+
+### June 8th - June 21th
+Session Establishment Framework: Julia wrappers to provide the functions required to establish sockets and begin the handshake process between clients will be implemented in this period.
+
+### June 22th - July 5th
+Session Establishment Framework / Certificates: The full functionality of the OpenSSL handshake protocol will be implemented and work on the generation of X509. certificates will begin.
+
+### July 6th - July 19th
+Session Establishment Framework / Certificates: The wrappers for the programs required for interacting with certificate authorities and verifying the validity of certificates will be completed by the end of this period.
+
+### July 20th - August 2rd
+Data and Signal Module: The modules required to send and receive data and the corresponding information signals will be implemented. The functions will be wrapped from the openssl.h library.
+
+### August 3rd - August 16th
+juliaTLS API: The different modules will be combined to create Julia native atomic functions to provide TLS functionalitiy. A full TLS negotiation will be tested via HTTPS, with a simple set of methods (get, post). 
+
+### August 17th - August 21th 19:00 UTC
+
+**Week to scrub code, write tests, improve documentation, etc.**
+
+## Future works
+The juliaTLS package is intended to provide the basis upon which the secure Julia implementations of various application layer protocols can be built. The various cryptographic libraries may be extended to provide additional capabilities and functionality.
+
+## Open Source Development Experience
```

Comment:
Currently No. I had mentioned my lack of experience to Seth when I was discussing the project with him.
 
I have experience in working with the Web Stack but not native to Julia, but I have already started learning how to write wrappers from a few resources ( http://docs.julialang.org/en/latest/manual/calling-c-and-fortran-code/). I have also discussed the project with Amit Murthy who has developed the julia wrappers for the OpenSSL crypto library, and Seth is always helpful.